XMonad/Actions/CycleWindows.hs

```diff
@@ -205,7 +209,7 @@ rotFocusedDown = windows . W.modify' $ rotFocused' rotDown
 rotFocused' :: ([a] -> [a]) -> W.Stack a -> W.Stack a
 rotFocused' _ s@(W.Stack _ [] []) = s
 rotFocused' f   (W.Stack t [] (r:rs)) = W.Stack t' [] (r:rs') -- Master has focus
-    where (t':rs') = f (t:rs)
+    where (nonEmpty -> t' :| rs') = f (t:rs)
```

Comment:
`f` is an arbitrary function, so I guess this technically isn't safe.  Since the function is exported we should probably just return the original stack in case of an empty list here

Reply:
`f` being arbitrary, this can blow up in other ways, like losing windows, for example, which can happen silently and there's nothing we can do about it even if we wanted to, because there's no `Eq`/`Ord` here. :-/
I'd personally prefer to be verbose at least in the case of `f` returning `[]`, though. If someone passes a function that does more than just a reordering, they should be told.

Reply:
May I consider this and the other two "arbitrary `f`" comments dealt with or would you still like a silent fallback rather than an exception?

Reply:
It's probably fine.  Not like anyone has ever complained about these functions blowing up so it doesn't seem to be a problem (and if it does turn out to be you're right that exceptions will make this easier to debug)